### PR TITLE
Remove dependency on `is-terminal` crate

### PR DESCRIPTION
**Pull request by Kmeakin**
The same functionality as `is_terminal::is_terminal` is now provided by `std::io::IsTerminal::is_terminal()`

### CODE REVIEW

**Comment by Pat-Lafon** [comment]
So is this just waiting for 1.72 to come out to update the msrv to 1.70?

**Comment by Kmeakin** [comment]
> So is this just waiting for 1.72 to come out to update the msrv to 1.70?

Yes

**Comment by Pat-Lafon** [comment]
@Kmeakin I think you just need to update https://github.com/lalrpop/lalrpop/blob/3d5195b454ec8f64c911afd35ecd997c77231e52/Cargo.toml#L23 and https://github.com/lalrpop/lalrpop/blob/3d5195b454ec8f64c911afd35ecd997c77231e52/.github/workflows/test.yaml#L19

**Review by yannham — APPROVED** [review]
Thanks, one less dependency :slightly_smiling_face:  LGTM. @youknowone do we want to block on that a bit, as 1.70 is quite close to the current Rust stable version? But I suspect we'll see a version bump before we release anything, and `n-2` was one of the original suggestion as a MSRV. I'm fine either way, merging this right now or later

**Comment by Pat-Lafon** [comment]
@yannham FWIW I don't think this should be blocked only because this waited until now to be within the msrv policy that was set with Niko's input. In https://github.com/lalrpop/lalrpop/issues/742 I believe we hashed out `N-2` as an appropriate amount of time to wait before moving up the msrv. There doesn't seem to be a reason to wait any longer here.

My understanding of trying to make that policy is that the maintainers didn't want to rehash this issue every time it comes up that the msrv value could be changed.

**Comment by youknowone** [comment]
I don't have strong opinion.

My weak idea is this patch is not adding feature but keeping compatibility while bumping rust version. In other words, one more dependency to keep 6 months release compatibility looks good deal to me unlike making a new feature patch in pain way. Even more, it will be really easy to rebase later. So nothing to lose.

Still, I don't have strong opinion about it. Please go ahead if anyone think this is good to go right now.

**Comment by yannham** [comment]
Thanks for your input both, I think it's reasonable to proceed.

**Pull request merged**
